### DISCUSSION TITLE
Easier expand/collapse for matrix blocks

### DISCUSSION THREAD

**Discussion by acalvino4** [craftcms/cms#10240] [Ideas] [answered]
I think Matrix blocks could use a better UI for expanding and collapsing. It just doesn't seem like the kind of thing that should take two clicks (one to open the menu, and another to press expand). There really is plenty of space in the matrix block header bar for an expand/collapse toggle).

Besides this, it would be really nice to have a select/deselect all checkbox toggle at the top of the matrix field, instead of having to select/deselect each block individually.

![image](https://user-images.githubusercontent.com/43254861/145852162-bc97771f-0ab0-4c83-aff4-9fa2f690b8d4.png)

**Comment by brandonkelly** [comment, marked as answer]
Agree it might make sense to add a dedicated toggle to the toolbar, and will consider this for a future release.

Couple tips that might help in the meantime, though:

- If multiple blocks are selected (via their checkboxes), the Expand/Collapse menu option will toggle them all simultaneously.
- You can quickly select all blocks by selecting one, then pressing <kbd>Ctrl</kbd>/<kbd>Command</kbd> + <kbd>A</kbd>, or selecting a range of blocks by selecting one, and <kbd>Shift</kbd>-clicking on the last one in the range.
- You can quickly toggle an individual block by double-clicking on its header.

**Reply by acalvino4** [reply]
Thanks Brandon, those shortcuts are super helpful!

I think that while those shortcuts are great for devs, most clients will probably never learn or remember to use them, so more obvious ways of doing those actions would still be nice.
